Tests/AmplitudeTests/Utilities/HttpClientTests.swift

```diff
@@ -42,6 +42,23 @@ final class HttpClientTests: XCTestCase {
         }
     }
 
+    func testGetRequestData() {
+        let httpClient = HttpClient(configuration: configuration)
+        let event = BaseEvent(userId: "unit-test user", eventType: "unit-test event")
+
+        let dateFormatter = ISO8601DateFormatter()
+        let currentDate = Date()
```

Comment:
1. The test is unstable. It compares 2 `Date()` that can produce different values. Maybe it's better to add an explicit `Date` argument to `getRequestData()` function
2. `currentDate` should be hard-coded test value, the test should compare actual value with hard-coded string like `2023-10-19 05:34:53.363` and do not repeat Iso861 logic here.

Reply:
Would it be bad to add a `Date` argument to `getRequestData()` function as it's only used in this test? I didn't find how to mock `Date()`. Maybe we should just delete the test?

Reply:
@Mercy811 as discussed in sync, another option is to:
1. Define an internal `getDate()` method in HttpClient
2. Override getDate() in a testing subclass like `FakeHttpClient` to return a fixed value https://github.com/amplitude/Amplitude-Swift/blob/main/Tests/AmplitudeTests/Supports/TestUtilities.swift#L116
3. Use the fake client in tests

Reply:
I did a research on this and found that it's a standard way to inject the test value as a parameter of the function. Update it [here](https://github.com/amplitude/Amplitude-Swift/pull/94/commits/2be2931301e77263a1d72af64e5409beae8a0a06).

Reply:
Hi @justin-fiedler, I compared the way we discussed and the one to add `currentTime` as a parameter. I chose the latter one as it's easier to implement.

Reply:
Thanks @Mercy811. 

I agree that adding `currentTime` to `getRequestData()` is a smaller change but it has the following downsides:

1. It adds a new parameter to the method that is only needed for testing. I feel it would be nice to keep this hidden from end users.
2. If we no longer need currentTime in HttpRequest it would require changing the method signature again
3. If we use currentTime elsewhere in HttpClient in the future, those methods would also need to add currentTime to their params

Since customers are unlikely to extend our HttpClient I think the currentTime solution is "ok" but would still prefer the internal getDate() method implementation.

Reply:
Hi @justin-fiedler, as we discussed, I use `getDate()` in `getRequestData()` and override `getDate()` in `FakeHttpClient` so that we can mock the current date. I really like the current solution and thank you for all the discussion. ❤️

Reply:
Thanks @Mercy811! Updates look great!